### ISSUE TITLE
[Question]: Applying GB-MMSA on backmapped coarse grained structures

### ISSUE THREAD

**Issue by srdjan125** [Valdes-Tresanco-MS/gmx_MMPBSA#225]
Hello,

I have performed MD simulations in SIRAH force field and then I have used their tools to backmap it back into all-atom trajectory. According to their SIRAH tools they backmap caoarse grained structure according to force field: $AMBERHOME/dat/leap/cmd/oldff/leaprc.ff14SB (that means it loads the ff14SB for proteins and ff99bsc0 for DNA).

I get output from backmaping as PDB file with N frames, and I can convert that file to .trr or .xtc trajectory format to use it as input for gmx_MMPBSA.

My .in file looks like this:
&general
sys_name="Prot-DNA"
startframe=1
endframe=20
PBRadii=4
forcefield = "oldff/leaprc.ff14SB"
/
&gb
igb=8, saltcon=0.150, intdiel=10
/

 I use this command:

**_gmx_MMPBSA -O -i mmpbsa.in -cs backmap.pdb -ci index.ndx -cg 1 12 -ct Traj.xtc  -cr backmap.pdb_**

However, when I try to start gmx_MMPBSA calculation, I get the following error:

**_Error: # atoms in XTC file (19310) does not match # atoms in parm COM.prmtop (19301)
Error: Could not set up 'COM_traj_0.xtc' for reading.
Error: Could not set up input trajectory 'COM_traj_0.xtc'.
Error: Error(s) occurred during execution.
  File "/home/srdjan/anaconda3/envs/AmberTools21/bin/gmx_MMPBSA", line 8, in <module>
    sys.exit(gmxmmpbsa())
  File _"/home/srdjan/anaconda3/envs/AmberTools21/lib/python3.9/site-packages/GMXMMPBSA/app.py",_ line 97, in gmxmmpbsa
    app.file_setup()
  File "/home/srdjan/anaconda3/envs/AmberTools21/lib/python3.9/site-packages/GMXMMPBSA/main.py", line 145, in file_setup
    self.numframes, self.numframes_nmode = make_trajectories(INPUT, FILES, self.mpi_size,
  File "/home/srdjan/anaconda3/envs/AmberTools21/lib/python3.9/site-packages/GMXMMPBSA/make_trajs.py", line 68, in make_trajectories
    traj = Trajectory(FILES.complex_prmtop, FILES.complex_trajs, cpptraj)
  File "/home/srdjan/anaconda3/envs/AmberTools21/lib/python3.9/site-packages/GMXMMPBSA/make_trajs.py", line 471, in __init__
    self.Query()
  File "/home/srdjan/anaconda3/envs/AmberTools21/lib/python3.9/site-packages/GMXMMPBSA/make_trajs.py", line 606, in Query
    raise TrajError('%s failed when querying %s' % (self.exe, traj))**_

  
As I understand, the problem might be that gmx_MMPBSA/AmberTools is removing protein and ligand from H-atoms and then adding them again. So at the end number of hydrogen might be different than in my .xtc file and the program crushes.
   Maybe also some backmapped atom atom types are not recognized by AmberTools.

 I would appreciate if somebody could help me with this issue.

Thank you in advance.

**Comment by marioernestovaldes** [comment]
Hi there! exactly... that seems to be the problem... I assume you don't get an all-atom topology of the back mapped system (if so the easiest way could be to define the topology with the -cp flag)... could you please send the files you are using to see if we can spot the problem?

**Comment by srdjan125** [comment]
Hi , thank you for your help.

Unfortunately, I don't have topology files, I have topology files of original GROMACS SIRAH topology, and only all-atom backed according to Amber force field.
I need to check if I can supply some option for -cp   I think that SIRAH can create psf file for all-atom but I don't know if gmx_mmpbsa has option to accept psf file.

   Here I am attaching the files which I have used. I also had the issue when I tried to use file backmap_original.pdb that AmberTools could not recognise OXT atoms at C-ter of 6 monomers, i don't understand why because sirah_backmap should create atom names according to Amber naming scheme. (After googling I found out that those are oxygens  at COO groups) I have found the ugly advice on forums to delete those atoms and let AmberTools rebuild any missing residue.

I guess the workout is to use parameters created by AmberTools and then strip my trajectory of hydrogens and protonate them again according to AmberTools scheme. But then I loose SIRAH advantage that I can use PQR files according to pH, while gromacs and AmberTools only protonate at pH 7. (Although in this case I have Propka structure according to pH 7 so the difference shouid be small

________________________________
